src/Compilers/VisualBasic/Portable/Lowering/LocalRewriter/LocalRewriter.vb

```diff
@@ -362,26 +365,25 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             End Get
         End Property
 
-        Private ReadOnly Property GenerateDebugInfo As Boolean
+        Private ReadOnly Property Instrument As Boolean
             Get
-                Return _globalGenerateDebugInfo AndAlso Not _inExpressionLambda
+                Return _instrumenter IsNot Instrumenter.NoOp AndAlso Not _inExpressionLambda
```

Comment:
It feels contradictory to both use the null object pattern (i.e. an instrumentor that doesn't do anything) and then also test for it to avoid calling it. I would expect to either use `Nothing` in place of an instrumentor (and test for nothing), or just call the no-op instrumenter and let it do its nothing.